birdhouse/deployment/deploy.sh

```diff
@@ -53,70 +53,75 @@
     usage
     exit 2
 else
     shift
 fi
 
-if [ -z "${ENV_LOCAL_FILE}" ]; then
-    ENV_LOCAL_FILE="${COMPOSE_DIR}/env.local"
-else
-    shift
+BIRDHOUSE_LOCAL_ENV="${1:-${BIRDHOUSE_LOCAL_ENV:-"${COMPOSE_DIR}/env.local"}}"
+
+# Setup COMPOSE_DIR and PWD for sourcing env.local.
+# Prevent un-expected difference when this script is run inside autodeploy
+# container and manually from the host.
+cd "${COMPOSE_DIR}" || exit
+
+. "${COMPOSE_DIR}/read-configs.include.sh"
+
+# Read BIRDHOUSE_AUTODEPLOY_EXTRA_REPOS
+read_basic_configs_only
+
+if [ ! -z "${AUTODEPLOY_SILENT}" ]; then
+    LOG_FILE="${BIRDHOUSE_LOG_DIR}/autodeploy.log"
+    mkdir -p "${BIRDHOUSE_LOG_DIR}"
+    exec >> "${LOG_FILE}" 2>&1
 fi
 
 COMPOSE_DIR="$(realpath "${COMPOSE_DIR}")"
 
+# This `if` block is required when upgrading from a version of the birdhouse-deploy code
+# without birdhouse-compose.sh to one with birdhouse-compose.sh. When pavics-compose.sh
+# is eventually deprecated and removed we can also remove this block.
+if [ -z "${BIRDHOUSE_COMPOSE}" ] && [ ! -f "${COMPOSE_DIR}/birdhouse-compose.sh" ]; then
+  BIRDHOUSE_COMPOSE="${COMPOSE_DIR}/pavics-compose.sh"
+fi
+
+BIRDHOUSE_COMPOSE=${BIRDHOUSE_COMPOSE:-"${COMPOSE_DIR}/birdhouse-compose.sh"}
+
 if [ ! -f "${COMPOSE_DIR}/docker-compose.yml" ] || \
-   [ ! -f "${COMPOSE_DIR}/pavics-compose.sh" ]; then
-    echo "ERROR: missing docker-compose.yml or pavics-compose.sh file in '${COMPOSE_DIR}'" 1>&2
+   [ ! -f "${BIRDHOUSE_COMPOSE}" ]; then
+    echo "ERROR: missing docker-compose.yml or birdhouse-compose.sh file in '${COMPOSE_DIR}'" 1>&2
     exit 2
 fi
 
-if [ ! -f "${ENV_LOCAL_FILE}" ]; then
-    echo "ERROR: env.local '${ENV_LOCAL_FILE}' not found, please instantiate from '${COMPOSE_DIR}/env.local.example'" 1>&2
+if [ ! -f "${BIRDHOUSE_LOCAL_ENV}" ]; then
+    echo "ERROR: env.local '${BIRDHOUSE_LOCAL_ENV}' not found, please instantiate from '${COMPOSE_DIR}/env.local.example'" 1>&2
     exit 2
 fi
 
 if [ -f "$COMPOSE_DIR/docker-compose.override.yml" ]; then
-    echo "WARNING: docker-compose.override.yml found, should use EXTRA_CONF_DIRS in env.local instead"
+    echo "WARNING: docker-compose.override.yml found, should use BIRDHOUSE_EXTRA_CONF_DIRS in env.local instead"
 fi
 
-# Setup COMPOSE_DIR and PWD for sourcing env.local.
-# Prevent un-expected difference when this script is run inside autodeploy
-# container and manually from the host.
-cd "${COMPOSE_DIR}" || exit
-
 START_TIME="$(date -Isecond)"
 echo "deploy START_TIME=${START_TIME}"
 
-. "${COMPOSE_DIR}/read-configs.include.sh"
-
-# Read AUTODEPLOY_EXTRA_REPOS
-read_basic_configs_only
-
 set -x
 
-for adir in "${COMPOSE_DIR}" ${AUTODEPLOY_EXTRA_REPOS}; do
+for adir in "${COMPOSE_DIR}" ${BIRDHOUSE_AUTODEPLOY_EXTRA_REPOS}; do
     if [ -d "${adir}" ]; then
         cd "${adir}" || exit
 
         # fail fast if unclean checkout
-        if [ ! -z "$(git status -u --porcelain)" ]; then
+        if [ ! -z "$(git status --untracked-files=no --porcelain)" ]; then
```

Comment:
This should be `--untracked-files=all` because the default was `all`.

Reply:
Yes that is the old default. But why? I mean, does it break the update if there are untracked files?

Reply:
It's not that it would break the autodeploy, it's more about if the checkout is unclean, then someone is experimenting with the machine, let's not break his test environment.

In other works, autodeploy can also be enabled on dev/staging machine and it will automatically keep those machines up-to-date as long as the machine is not being actively used for development purposes.  Very convenient when you have a fleet of test machines and you want them to be always up-to-date, ready to be used as test environment, and when they are in used, autodeploy won't touch them.

On a production machine, having untracked files means someone is testing directly in prod !!! It's probably for an emergency, again do not break that. And it gives a very simple and quick way to temporary disable autodeploy: just touch a new file like `disable-autodeploy-because-of -XYZ-reason.txt`.

Reply:
I'm going to revert this because making this change here was the wrong place to do it. 

> On a production machine, having untracked files means someone is testing directly in prod

There are many other reasons why you might have untracked files in prod:

- multiple env.local files
- custom components 
- notes and readme files 

By restricting what can go in this folder we limit lots of useful use cases. This is one of the reasons that I don't use the auto deploy components in production (the other being that I want to be able to understand updates before I deploy them in production).